pip_audit/_service/osv.py

```diff
@@ -110,10 +112,10 @@ def query(self, spec: Dependency) -> Tuple[Dependency, List[VulnerabilityResult]
             # formatting in the process).
             description = description.replace("\n", " ")
 
-            aliases = set(vuln.get("aliases", []))
-
             # OSV doesn't mandate this field either. There's very little we
             # can do without it, so we skip any results that are missing it.
+            #
+            # TODO(@orsinium): not true anymore, the field is marked as required in API docs.
```

Comment:
Let's just revise this comment instead.

Reply:
@orsinium where did you see that this field is marked as required?

Here's what I see in the OSV schema docs (https://ossf.github.io/osv-schema/#id-modified-fields):

> The id and modified fields are required. All other fields are optional, although of course an entry with no other metadata is not particularly useful. (It could potentially stand for a reserved ID with no other public information.)

Reply:
(It makes sense for `affected` in particular to be optional, for the "reserved ID" use case mentioned in the docs.)

Reply:
In [API docs](https://osv.dev/docs/#tag/api/operation/OSV_QueryAffected) for 200 Response fields:

> affected | Array of objects (osvAffected)
> **Required**. Affected commit ranges and versions.

Reply:
This is a funny case: OSV's own API might alway return the `affected` key, but the OSV specification says that it's optional. I'm going to open an upstream issue to get clarification here.

Reply:
Ref: https://github.com/google/osv.dev/issues/810